### ISSUE TITLE
[max-params] can ignore constructor() of Class?

### ISSUE THREAD

**Issue by waitingsong** [eslint/eslint#13090]
**What rule do you want to change?**
[max-params](https://eslint.org/docs/rules/max-params)

**Does this change cause the rule to produce more or fewer warnings?**

**How will the change be implemented? (New option, new default behavior, etc.)?**
Provide a flag, it can ignore the constructor() of Class

**Please provide some example code that this change will affect:**

```ts
class Foo {
  constructor(
    input1: string,
    input2: string,
    input3: string,
    input4: string,
  ) {
  // ...
  }
}

```

**What does the rule currently do for this code?**
```
error    Constructor has too many parameters (4). Maximum allowed is 3  max-params
````

**What will the rule do after it's changed?**
No error 

**Are you willing to submit a pull request to implement this change?**

**Comment by anikethsaha** [comment]
you can use `/*eslint max-params: ["error", 4]*/` to make it lint-free. It will be applied for all the functions across the code. 

_Though, an option like `whiteList` to add the name of the functions/methods can be passed which will be ignored can be considered. Not sure, let's hear from the others_

**Comment by waitingsong** [comment]
> you can use `/*eslint max-params: ["error", 4]*/` to make it lint-free. It will be applied for all the functions across the code.
> 
> _Though, an option like `whiteList` to add the name of the functions/methods can be passed which will be ignored can be considered. Not sure, let's hear from the others_

thanks.

**Comment by kaicataldo** [comment]
Closing this since the question has been answered. @anikethsaha Thanks!
